apps/agent/test/agent_shared.js

```diff
@@ -172,6 +194,72 @@ module.exports = (
         await assertRevert(agent.initialize(), errors.INIT_ALREADY_INITIALIZED)
       })
 
+      context('> ERC165', () => {
+        it('supports base ERC165 interface', async () => {
+          assert.isTrue(await agent.supportsInterface(ERC165_SUPPORT_INTERFACE_ID))
+          assert.isFalse(await agent.supportsInterface(ERC165_SUPPORT_INVALID_ID))
+        })
+
+        it('supports ERC1271 interface', async () => {
+          assert.isTrue(await agent.supportsInterface(ERC1271_INTERFACE_ID))
+        })
+
+        it('supports ERC721Receiver interface', async () => {
+          assert.isTrue(await agent.supportsInterface(ERC721_RECEIVED_INTERFACE_ID))
+        })
+
+        it('supports ERC1155Receiver interface', async () => {
+          assert.isTrue(await agent.supportsInterface(ERC1155_RECEIVER_INTERFACE_ID))
+        })
+
+        it("doesn't support any other interface", async () => {
+          assert.isFalse(await agent.supportsInterface('0x12345678'))
+          assert.isFalse(await agent.supportsInterface('0x'))
+        })
+      })
+
+      context('> ERC721', () => {
+        it('supports ERC721 receiver callback', async () => {
+          const [token, operator, from] = accounts
+          const tokenId = 5
+          const data = '0x12345678'
+
+          const callbackReturn = await agent.onERC721Received.call(operator, from, tokenId, data)
+          assert.equal(callbackReturn, ERC721_RECEIVED_INTERFACE_ID, 'expected ERC721 receipt magic return')
+
+          const receipt = await agent.onERC721Received(operator, from, tokenId, data)
+          assertAmountOfEvents(receipt, 'ReceiveERC721')
+        })
+      })
+
+      context('> ERC1155', () => {
+        it('supports ERC1155 single receipt callback', async () => {
+          const [token, operator, from] = accounts
+          const id = 5
+          const value = 10
+          const data = '0x12345678'
+
+          const callbackReturn = await agent.onERC1155Received.call(operator, from, id, value, data)
+          assert.equal(callbackReturn, ERC1155_SINGLE_RECEIVED_INTERFACE_ID, 'expected ERC1155 single receipt magic return')
+
+          const receipt = await agent.onERC1155Received(operator, from, id, value, data)
+          assertAmountOfEvents(receipt, 'ReceiveERC1155')
```

Comment:
We can add a second assert here to check the logged values

Reply:
Was thinking that... but then stopped because we haven't done this elsewhere 😅.

Agree that we should definitely do this though; do the new contracts include tests for the event parameters?

Reply:
I think we should, otherwise this is untested logic. In fact, I always do test the event params haha